soc/xtensa/intel_adsp/ace_v1x/power.c

```diff
@@ -43,7 +46,18 @@ __weak void pm_state_set(enum pm_state state, uint8_t substate_id)
 		if (cpu == 0) {
 			/* FIXME: this value should come from MM */
 			uint32_t hpsram_mask[1] = { 0x3FFFFF };
+#if CONFIG_PM_DEVICE
+			const struct device *dev = device_get_binding("HPSRAM_TLB");
 
+			if (dev != NULL) {
+				/* the TLB driver suspen will save the whole memory
+				 * leaving the system not operational
+				 * it must be called directly here,
+				 * just before ace_power_down
+				 */
+				pm_device_action_run(dev, PM_DEVICE_ACTION_FORCE_SUSPEND);
+			}
```

Comment:
This is not correct. If you want to suspend / resume devices manually you have to use device runtime API.

Now the most important thing, devices are automatically suspended / resumed by the power subsystem if they are not busy or doing / using device runtime pm.

Reply:
I was under impression that Zephyr won't call it automatically unless it is enabled for a device:

bool pm_device_runtime_is_enabled(const struct device *dev)
{
	struct pm_device *pm = dev->pm;

	return pm && atomic_test_bit(&pm->flags, PM_DEVICE_FLAG_RUNTIME_ENABLED);
}

Can't let Zephyr do it automatically, it is too specific/sensitive and must be called right before "turning the light off"....
Changing it to specific API

Reply:
so, you have to use device_runtime_pm for this device as I'd suggested. `pm_device_action_run` is low level function, if you are manually suspend/resuming a device you should use device runtime -> https://docs.zephyrproject.org/latest/services/pm/device_runtime.html